server/eclipse-project/src/main/java/us/freeandfair/corla/endpoint/AbstractDoSEndpoint.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Free & Fair Colorado RLA System
+ * 
+ * @title ColoradoRLA
+ * @created Aug 12, 2017
+ * @copyright 2017 Free & Fair
+ * @license GNU General Public License 3.0
+ * @creator Joe Kiniry <kiniry@freeandfair.us>
+ * @description A system to assist in conducting statewide risk-limiting audits.
+ */
+
+package us.freeandfair.corla.endpoint;
+
+import us.freeandfair.corla.asm.DoSDashboardASM;
+
+/**
+ * Functionality that spans endpoints on the Department of State Dashboard.
+ */
+@SuppressWarnings("PMD.AtLeastOneConstructor")
+public abstract class AbstractDoSEndpoint extends AbstractEndpoint {
+  /**
+   * @return State authorization is required for these endpoints.
+   */
+  @Override
+  public AuthorizationType requiredAuthorization() {
+    return AuthorizationType.STATE;
+  }
+  
+  /**
+   * @return subclasses of this endpoint use the Department of State ASM.
+   */
+  @Override
+  protected Class<DoSDashboardASM> asmClass() {
+    return DoSDashboardASM.class;
+  }
```

Comment:
I think we need another method here, something like `protected String asmIdentity(final Request the_request)` that returns the ASM's identity string; for the DoS dashboard it'd return `null`, because that's a singleton (we could also choose something else consistent other than `null`), but for the others, it would check the request and extract the county ID that the county administrator is authenticated as (using `Authentication.authenticatedCounty(Request)`).

There may be other ways to do this too.